File: w-gory-java/src/main/java/com/wGory/model/TrasaPunktowana.java
package com.wGory.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.Entity;
import javax.persistence.OneToOne;
import javax.persistence.PrimaryKeyJoinColumn;
import javax.persistence.Table;

@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "TRASA_PUNKTOWANA")
@Data
@PrimaryKeyJoinColumn(name = "trasa_punktowana_id")
public class TrasaPunktowana extends OdcinekTrasy {
    @OneToOne
    private Miejsce poczatek;
    @OneToOne
    private Miejsce koniec;
    private Integer punkty;
}

File: w-gory-angular/src/app/model/trasaPunktowana.ts
import { Miejsce } from './miejsce';
import { OdcinekTrasy } from './odcinekTrasy';

export class TrasaPunktowana implements OdcinekTrasy{
    id: number;
    punkty: number;
    
    poczatek: Miejsce;
    koniec: Miejsce;
}